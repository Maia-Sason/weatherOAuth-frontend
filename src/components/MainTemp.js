import { Typography, makeStyles, Box } from "@material-ui/core";
import WeatherIcon from "./WeatherIcon";

const useStyles = makeStyles((themes) => ({
  root: {
    display: "flex",
  },
  headerB: {
    paddingLeft: "10px",
    fontSize: "7em",
    fontWeight: 100,
    color: "white",
    letterSpacing: "5px",
  },
  aside: {
    display: "inline",
    fontSize: "36px",
    color: "white",
    verticalAlign: "middle",
    letterSpacing: "2px",
    fontWeight: 200,
  },
  icon: {
    fill: "white",
    height: "6em",
    width: "6em",
    verticalAlign: "middle",
  },
}));

const MainTemp = ({ temp, city, icon }) => {
  const classes = useStyles();

  return (
    <>
      <div className={classes.root}>
        <Typography className={classes.headerB}>
          {temp.toFixed(0)}&deg;
          <Typography className={classes.aside}>{city}</Typography>
        </Typography>
        <Box className={classes.icon}>
          <WeatherIcon icon={icon}></WeatherIcon>
        </Box>
      </div>
    </>
  );
};

export default MainTemp;
