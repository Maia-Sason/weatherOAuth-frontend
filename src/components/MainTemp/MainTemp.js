import { Typography, makeStyles, Box } from "@material-ui/core";
import WeatherIcon from "../WeatherIcon";

const useStyles = makeStyles((themes) => ({
  root: {
    display: "flex",
    direction: "column",
    contentAlign: "center",
  },
  headerB: {
    paddingLeft: "10px",
    fontSize: "7em",
    fontWeight: 100,
    color: "white",
    letterSpacing: "5px",
  },
  info: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
  },
  aside: {
    display: "inline",
    fontSize: "36px",
    color: "white",
    verticalAlign: "bottom",
    letterSpacing: "2px",
    fontWeight: 200,
  },
  icon: {
    fill: "white",
    height: "4em",
    width: "4em",
  },
}));

const MainTemp = ({ temp, city, icon }) => {
  const classes = useStyles();

  return (
    <>
      <div className={classes.root}>
        <Typography className={classes.headerB}>
          {temp.toFixed(0)}&deg;
        </Typography>
        <div className={classes.info}>
          <Box className={classes.icon}>
            <WeatherIcon icon={icon}></WeatherIcon>
          </Box>
          <Typography className={classes.aside}>{city}</Typography>
        </div>
      </div>
    </>
  );
};

export default MainTemp;
