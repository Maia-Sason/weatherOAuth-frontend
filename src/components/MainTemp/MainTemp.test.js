import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import MainTemp from "./MainTemp";

test("<MainTemp /> matches snapshot", () => {
  const component = render(<MainTemp temp={12.12} />);
  expect(component.container).toMatchSnapshot();
});

it("renders MainTemp without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MainTemp temp={12.12} />, div);
});
