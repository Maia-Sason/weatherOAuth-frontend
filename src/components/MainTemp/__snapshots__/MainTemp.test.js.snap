// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainTemp /> matches snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <p
      class="MuiTypography-root makeStyles-headerB-2 MuiTypography-body1"
    >
      12
      Â°
    </p>
    <div
      class="makeStyles-info-3"
    >
      <div
        class="MuiBox-root MuiBox-root-6 makeStyles-icon-5"
      />
      <p
        class="MuiTypography-root makeStyles-aside-4 MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
